### PR TITLE
RWA-452 Implement Provider Contract/Pact Test - POST /workflow/message

### PR DESCRIPTION
**Pull request by mkayad**
### JIRA link (if applicable) ###

[RWA-452](https://tools.hmcts.net/jira/browse/RWA-452)

### Change description ###

see ticket

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -51,4 +56,16 @@ withPipeline(type, product, component) {
     steps.archiveArtifacts allowEmptyArchive: true, artifacts: '**/site/serenity/**/*'
     steps.archiveArtifacts allowEmptyArchive: true, artifacts: '**/reports/tests/functional/**/*'
   }
+
+//  after('pact-provider-verification') {
+//    steps.archiveArtifacts allowEmptyArchive: true, artifacts: 'build/reports/tests/**/*'
+//  }
```
Comment (alexbaeza) [inline Jenkinsfile_CNP:62]: We could remove this

**Review by alexbaeza — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -248,7 +263,7 @@ repositories {
 }
 
 def versions = [
-  junit           : '5.7.1',
+  junit           : '5.7.0',
```
Comment (iguannago) [inline build.gradle:266]: is this needed??

**Review by indvamshi — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (mkayad) [reply build.gradle]: cancelled the downgrade

File: Jenkinsfile_CNP (hunk 1 above)
Comment (mkayad) [reply Jenkinsfile_CNP]: code removed

File: src/contractTest/java/uk/gov/hmcts/reform/waworkflowapi/consumer/wa/WorkflowPostMessageConsumerTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import uk.gov.hmcts.reform.waworkflowapi.provider.service.CamundaConsumerApplication;
+
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.waworkflowapi.clients.model.DmnValue.dmnStringValue;
+
+@PactTestFor(providerName = "wa_workflow_api_post_message", port = "8899")
+@ContextConfiguration(classes = {CamundaConsumerApplication.class})
+@TestPropertySource(locations = {"classpath:application.properties"})
+public class WorkflowPostMessageConsumerTest extends SpringBootContractBaseTest {
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/waworkflowapi/consumer/wa/WorkflowPostMessageConsumerTest.java:28]: Can we update this consumer name to match the name of the method on the CreateTaskController class? I.e. sendMessage

File: src/contractTest/java/uk/gov/hmcts/reform/waworkflowapi/provider/WorkflowPostMessageProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+@PactBroker(
+    scheme = "${PACT_BROKER_SCHEME:http}",
+    host = "${PACT_BROKER_URL:localhost}",
+    port = "${PACT_BROKER_PORT:9292}",
+    consumerVersionSelectors = {
+        @VersionSelector(tag = "master")}
+)
+@Import(WorkflowProviderTestConfiguration.class)
+@IgnoreNoPactsToVerify
+public class WorkflowPostMessageProviderTest {
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/waworkflowapi/provider/WorkflowPostMessageProviderTest.java:48]: Can we update this consumer name to match the name of the method on the CreateTaskController class? I.e. WorkflowSendMessageProvider

File: src/contractTest/java/uk/gov/hmcts/reform/waworkflowapi/consumer/wa/WorkflowPostMessageConsumerTest.java (hunk 3 above)
Comment (mkayad) [reply src/contractTest/java/uk/gov/hmcts/reform/waworkflowapi/consumer/wa/WorkflowPostMessageConsumerTest.java]: class renamed

File: src/contractTest/java/uk/gov/hmcts/reform/waworkflowapi/consumer/wa/WorkflowSendMessageConsumerTest.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import uk.gov.hmcts.reform.waworkflowapi.SpringBootContractBaseTest;
+import uk.gov.hmcts.reform.waworkflowapi.clients.model.SendMessageRequest;
+import uk.gov.hmcts.reform.waworkflowapi.provider.service.CamundaConsumerApplication;
+
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.waworkflowapi.clients.model.DmnValue.dmnStringValue;
+
+@PactTestFor(providerName = "wa_workflow_api_send_message", port = "8899")
+@ContextConfiguration(classes = {CamundaConsumerApplication.class})
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/waworkflowapi/consumer/wa/WorkflowSendMessageConsumerTest.java:26]: are these used?
@ContextConfiguration(classes = {CamundaConsumerApplication.class})
@TestPropertySource(locations = {"classpath:application.properties"})

Comment (mkayad) [reply src/contractTest/java/uk/gov/hmcts/reform/waworkflowapi/consumer/wa/WorkflowSendMessageConsumerTest.java]: Tests run without it , so removed from the code, good catch :-)

**Pull request merged**